File: runtime/src/main/java/com/siimkinks/sqlitemagic/SqlClause.java
package com.siimkinks.sqlitemagic;

import com.siimkinks.sqlitemagic.internal.SimpleArrayMap;

import java.util.LinkedList;

import androidx.annotation.NonNull;

abstract class SqlClause {
  abstract void appendSql(@NonNull StringBuilder sb);

  abstract void appendSql(@NonNull StringBuilder sb, @NonNull SimpleArrayMap<String, LinkedList<String>> systemRenamedTables);
}

File: runtime/src/main/kotlin/com/siimkinks/sqlitemagic/SqlClause.kt
package com.siimkinks.sqlitemagic

import com.siimkinks.sqlitemagic.internal.SimpleArrayMap
import java.util.LinkedList

abstract class SqlClause {
  abstract fun appendSql(
    sb: StringBuilder
  )

  abstract fun appendSql(
    sb: StringBuilder,
    systemRenamedTables: SimpleArrayMap<String, LinkedList<String>>
  )
}